core/controller/cluster/controller_basic_v2.go

```diff
@@ -529,12 +530,11 @@ func (c *controller) customizeCreateReqBuildTemplateInfo(ctx context.Context, r
 
 	var appGitRepoFile *appgitrepo.GetResponse
 	var err error
-	if (r.BuildConfig != nil || r.TemplateInfo != nil) && mergePatch {
-		appGitRepoFile, err = c.applicationGitRepo.GetApplication(ctx, application.Name, environment)
-		if err != nil {
-			return nil, err
-		}
+	appGitRepoFile, err = c.applicationGitRepo.GetApplication(ctx, application.Name, environment)
```

Comment:
why remove `if (r.BuildConfig != nil || r.TemplateInfo != nil) && mergePatch`

Reply:
To fix nil point panic.
appGitRepoFile will be used in many other conditions.